xmtp/src/conversations.rs

```diff
@@ -176,17 +329,25 @@ where
                 };
             }
             None => {
-                (_, plaintext) = match self.client.create_inbound_session(
-                    conn,
-                    &invitation.inviter,
-                    &invitation.ciphertext,
-                ) {
-                    Ok((session, plaintext)) => (session, plaintext),
-                    Err(err) => {
-                        log::error!("Error creating session: {:?}", err);
+                let prek_key = match olm_message {
```

Comment:
OK. So, we are expecting that there will either be an existing session or you will be receiving a prekey message. 

We might have to revisit this if we run into cases where the first reply isn't delivered and one side is sending regular messages but the other side doesn't have an active session. Should be a pretty fringe edge case.

Reply:
> one side is sending regular messages but the other side doesn't have an active session

Just want to clarify this. An initiator will create a session and keep sending prekey messages until a reply is received. The only case where the initiator will send a non-prekey message is if the peer has replied, in which case the peer should have a session?

Reply:
Correct the processing flow has been updated to:

Is there a Session:
- Yes: Decrypt using existing session
- No: is this a PrekeyMessage:
     - YES: create inbound session
     - No: Error